src/foam/nanos/auth/FilterBySpidService.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+   package: 'foam.nanos.auth',
+   name: 'FilterBySpidService',
+   extends: 'foam.nanos.auth.ProxyUserLocatorService',
+   flags: ['java'],
+
+  documentation: `Restrict login to the url that matches the spid of the user.`,
+
+
+  javaImports: [
+    'foam.dao.DAO',
+    'foam.nanos.auth.AuthenticationException',
+    'foam.nanos.auth.User',
+    'foam.nanos.theme.Theme',
+
+    'static foam.mlang.MLang.*'
+  ],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'superSpid',
+      documentation: 'Set spid of a theme that is accessible to all users',
+      value: "",
+    }
+  ],
+
+  methods: [
+    {
+      name: 'getUser',
+      documentation: 'Helper logic function to reduce code duplication',
+      type: 'User',
+      args: [
```

Comment:
FYI: when defining args:, you can just do: args: 'Context x, String identifier, String password'

But maybe not required here anyway.

Reply:
yes, not required, as specified by the interface model.

Reply:
yes, right! I removed args, types, and documentation here.